### DISCUSSION TITLE
Help me please: ESP32 stuck in deep sleep (on T-Pico C3)

### DISCUSSION THREAD

**Discussion by FabriceAuz** [micropython/micropython#9829] [Hardware & Peripherals] [answered]
Hi
I need some help: the ESP32 of my [T-Pico C3](https://github.com/Xinyuan-LilyGO/T-PicoC3) is stuck in deep sleep.

I was testing the deep sleep feature of the ESP32. I planned to wake it up from the RP2040 with a signal on GPIO4.
My idea was to use the following sketch on the ESP32:

```
import esp32
from machine import Pin, UART
from machine import deepsleep
import time

# set UART
RP2040 = UART(1, baudrate=38400, tx=7, rx=6, cts=5, rts=4)
time.sleep_ms(50)

print('going to sleep')
wake1 = Pin(4, mode = Pin.IN)
esp32.wake_on_ext0(pin = wake1, level = esp32.WAKEUP_ANY_HIGH)
deepsleep()

time.sleep(1)
RP2040.write('OK')
```
It goes into deep sleep and wakes up on receiving a signal on the GPIO4 (rts pin of the UART). Then it would send a message to the RP2040 to tell it's awake. To run it automatically, I saved it as 'Sleep_ESP32.py' and run it using the main.py:
import Sleep_ESP32

On the RP2040 side, I run the following sketch:

```
# RP2040 side: test wake up ESP32
from machine import UART, Pin
from time import sleep_ms
# Init UART
ESP32 = UART (1, baudrate=38400, tx=Pin(8), rx=Pin(9), cts=Pin(10), rts=Pin(11))
PinWakeUp  = Pin (10)
# Init red LED
led = Pin(25, Pin.OUT)
led.off()

sleep_ms(2000)
print('sending wake up')
PinWakeUp.on()
sleep_ms(30)
PinWakeUp.off()

line = ''
while 1:
    while ESP32.any() > 0:
        led.on()
        buf = ESP32.readline()
        line += buf.decode("utf-8")
    if len(line) != 1: print(line)
```

It wakes for 2 seconds then sends a signal on the RTS pin (10), which is supposedly connected to the RTS pin (4) of the ESP32. If it receives a message from the UART, it lits the red led.

When I run it, with the ESP32 part automatically running also, I see the red led. So I guess the ESP32 is awaken.
But when I connect the board on the ESP32 side (green led on) I can't get any access to the USB port. The board does not appear in the device manager any more. I can't access it with Thonny either and can't access to the flash as a USB device.

It seems stuck in deep sleep. How can I access the ESP32 again. Is there a safe boot mode?
Or do I need to erase the flash and upload the firmware again?
Thanks for your help.

**Comment by FabriceAuz** [comment, marked as answer]
OK, I erased the flash and uploaded the firmware again

**Reply by puppet13th** [reply]
currently wake_pin have not implemented for esp32-c3. #9583
